dds/DCPS/XTypes/External.h

```diff
@@ -48,6 +48,8 @@ class External {
 
   bool operator==(const External& other) const { return ptr_ == other.ptr_; }
   bool operator!=(const External& other) const { return ptr_ != other.ptr_; }
+  bool operator<(const External& other) const { return ptr_ < other.ptr_; }
+  bool operator>(const External& other) const { return ptr_ > other.ptr_; }
```

Comment:
Is `operator>` needed for any of these?  The idiom we inherit from the standard library is to use `operator<` for associative container comparator.  If you need `a > b` you can write `b < a`.

Reply:
I believe this was a result of me using `>` operator in `TypeObject.h` functions, like:

    bool operator<(const PlainMapLTypeDefn& other) const
    {
      if (header < other.header) return true;
      if (header > other.header) return false;
      if (bound < other.bound) return true;
      if (bound > other.bound) return false;
      if (element_identifier < other.element_identifier) return true;
      if (element_identifier > other.element_identifier) return false;
      if (key_flags < other.key_flags) return true;
      if (key_flags > other.key_flags) return false;
      if (key_identifier < other.key_identifier) return true;
      if (key_identifier > other.key_identifier) return false;
      return false;
    }
  };

If I rewrite them without using `>` operators this one may not be needed.
Would you like me to do that?

Reply:
Implemented as suggested